repose-aggregator/components/filters/herp/src/main/scala/org.openrepose.filters.herp/HerpFilter.scala

```diff
@@ -0,0 +1,135 @@
+    if (!initialized) {
+      logger.error("HERP filter has not yet initialized...")
+      servletResponse.asInstanceOf[HttpServletResponse].sendError(500)
+    } else {
+      logger.trace("HERP filter passing request...")
+      filterChain.doFilter(servletRequest, servletResponse)
+
+      logger.trace("HERP filter handling response...")
+      handleResponse(servletRequest.asInstanceOf[HttpServletRequest], servletResponse.asInstanceOf[HttpServletResponse])
+
+      logger.trace("HERP filter returning response...")
+    }
+  }
+
+  private def handleResponse(httpServletRequest: HttpServletRequest,
+                             httpServletResponse: HttpServletResponse) = {
+    def translateParameters(): Map[String, Array[Parameter]] = {
+      Option(httpServletRequest.getAttribute("http://openrepose.org/queryParams")) match {
+        case Some(parameters) => {
+          val parametersMap = parameters.asInstanceOf[java.util.Map[String, Array[String]]].asScala
+          parametersMap.mapValues((values :Array[String]) => values.map(Parameter(_))).toMap
+        }
+        case None => Map[String, Array[Parameter]]()
+      }
+    }
+
+    val templateValues = Map(
+      "userName" -> httpServletRequest.getHeader(OpenStackServiceHeader.USER_NAME.toString),
+      "impersonatorName" -> httpServletRequest.getHeader(OpenStackServiceHeader.IMPERSONATOR_NAME.toString),
+      "tenantID" -> httpServletRequest.getHeader(OpenStackServiceHeader.TENANT_ID.toString),
+      "roles" -> httpServletRequest.getHeaders(OpenStackServiceHeader.ROLES.toString).asScala.map(Role(_)).toArray,
+      "userAgent" -> httpServletRequest.getHeader(CommonHttpHeader.USER_AGENT.toString),
+      "requestMethod" -> httpServletRequest.getMethod,
+      "requestURL" -> Option(httpServletRequest.getAttribute("http://openrepose.org/requestUrl")).map(_.asInstanceOf[String]).orNull,
+      "parameters" -> translateParameters().asJava.entrySet(),
+      "timestamp" -> System.currentTimeMillis(),
+      "responseCode" -> httpServletResponse.getStatus,
+      "responseMessage" -> HttpStatusCode.fromInt(httpServletResponse.getStatus).name(),
+      // todo: "responseBody" -> ??? // wrap the request in mutable request to get the body
+      "guid" -> java.util.UUID.randomUUID.toString,
+      "serviceCode" -> serviceCode,
+      "region" -> region,
+      "dataCenter" -> dataCenter
+    )
+
+    val templateOutput: StringWriter = new StringWriter
+    mustacheTemplate.execute(templateOutput, templateValues.asJava)
+
+    herpLogger.info(templateOutput.toString)
+  }
+
+  override def destroy(): Unit = {
+    logger.trace("HERP filter destroying ...")
+    configurationService.unsubscribeFrom(config, this.asInstanceOf[UpdateListener[_]])
+
+    logger.trace("HERP filter destroyed.")
+  }
+
+  override def configurationUpdated(config: HerpConfig): Unit = {
+    def template :StringReader = {
+      var  templateText = config.getTemplate.getValue.trim
```

Comment:
What happened with the formatting here? Double space, colon prepended to the type...